core/unit_test/headers_self_contained/tstHeader.cpp

```diff
@@ -0,0 +1,15 @@
+#define KOKKOS_HEADER_TEST_STRINGIZE_IMPL(x) #x
+#define KOKKOS_HEADER_TEST_STRINGIZE(x) KOKKOS_HEADER_TEST_STRINGIZE_IMPL(x)
+
+#define KOKKOS_HEADER_TO_TEST \
+  KOKKOS_HEADER_TEST_STRINGIZE(KOKKOS_HEADER_TEST_NAME)
+
+// include header twice to see if the include guards are set correctly
+#include KOKKOS_HEADER_TO_TEST
+#include KOKKOS_HEADER_TO_TEST
+
+#if !defined(KOKKOS_MACROS_HPP)
+#error "This header does not include Kokkos_Macros.hpp"
+#endif
```

Comment:
Are you positive all headers should be including it?

Reply:
I think that is the general assumption.
Of course, there are some headers like `Kokkos_Timer.hpp` that don't depend on anything defined in `Kokkos_Macros.hpp` at all. On the other hand, including `Kokkos_Macros.hpp` is really cheap since there are no other includes, only preprocessor definitions and it is probably a good idea to compile everything anew anyway if that file changes.
That's just my opinion and I am happy to hear others and adapt if necessary (although it gets much more complicated then).